### PR TITLE
chore: split up DID creation operation

### PR DESCRIPTION
**Pull request by ntn-x2**
This PR fixes https://github.com/KILTprotocol/ticket/issues/1884 and fixes https://github.com/KILTprotocol/ticket/issues/1897 by simplifying the creation operation to only include an authentication key. No more complex logic on the input is required, and multiple components can be batched together if needed.

Benchmarks and test cases have been adjusted accordingly.

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [x] I have documented the changes (where applicable)
- [ ] Try to implement a different origin for DID creation operation that would allow batching
- [ ] Integrate PR feedback
- [ ] Run benchmarks

### CODE REVIEW

File: pallets/did/src/benchmarking.rs [hunk 1]
```diff
@@ -124,208 +132,93 @@ benchmarks! {
...
-		did_creation_details.new_key_agreement_keys = did_key_agreement_keys;
-		did_creation_details.new_attestation_key = Some(DidVerificationKey::from(did_public_att_key));
-		did_creation_details.new_delegation_key = Some(DidVerificationKey::from(did_public_del_key));
-		did_creation_details.new_service_details = service_endpoints.clone();
+		let did_creation_details = generate_base_did_creation_details::<T>(did_subject.clone(), submitter.clone());
 
-		let did_creation_signature = ed25519_sign(AUTHENTICATION_KEY_ID, &did_public_auth_key, did_creation_details.encode().as_ref()).expect("Failed to create DID signature from raw ed25519 signature.");
-	}: create(RawOrigin::Signed(submitter), Box::new(did_creation_details.clone()), DidSignature::from(did_creation_signature))
+		let origin = RawOrigin::Signed(submitter);
+		let did_creation_signature = DidSignature::from(ed25519_sign(AUTHENTICATION_KEY_ID, &did_public_auth_key, did_creation_details.encode().as_ref()).expect("Failed to create DID signature from raw ed25519 signature."));
```
Comment (Diiaablo95) [inline pallets/did/src/benchmarking.rs:146]: Tried to make all conversions, cloning, etc before the actual benchmark call.

File: pallets/did/src/did_details.rs [hunk 2]
```diff
@@ -618,7 +571,7 @@ pub trait DeriveDidCallAuthorizationVerificationKeyRelationship {
 /// A DID operation that wraps other extrinsic calls, allowing those
 /// extrinsic to have a DID origin and perform DID-based authorization upon
 /// their invocation.
-#[derive(Clone, Debug, Decode, Encode, PartialEq, TypeInfo)]
```
Comment (Diiaablo95) [inline pallets/did/src/did_details.rs:621]: So we can drop the requirement about `Debug` in the did `Config` trait.

File: pallets/did/src/lib.rs [hunk 3]
```diff
@@ -1011,14 +955,7 @@ pub mod pallet {
...
 		#[allow(clippy::boxed_local)]
-		#[pallet::weight({
-			let di = did_call.call.get_dispatch_info();
-			let max_sig_weight = <T as pallet::Config>::WeightInfo::submit_did_call_ed25519_key()
-			.max(<T as pallet::Config>::WeightInfo::submit_did_call_sr25519_key())
-			.max(<T as pallet::Config>::WeightInfo::submit_did_call_ecdsa_key());
-
-			(max_sig_weight.saturating_add(di.weight), di.class)
-		})]
+		#[pallet::weight(1)]
```
Comment (weichweich) [inline pallets/did/src/lib.rs:958]: Don't forget to add the weight.

File: runtimes/standalone/src/lib.rs [hunk 4]
```diff
@@ -395,6 +394,7 @@ impl did::Config for Runtime {
 	type DidIdentifier = DidIdentifier;
 	type Event = Event;
 	type Call = Call;
+	type CallProxy = DidProxy;
```
Comment (weichweich) [inline runtimes/standalone/src/lib.rs:397]: I'm a bit confused by the naming. What does this do?

**Comment by ntn-x2** [comment]
This has been moved back to draft to be improved, so please ignore the review request until I re-request it, when I will pick this up again. Thanks.

**Comment by ntn-x2** [comment]
Closed in favor of https://github.com/KILTprotocol/mashnet-node/pull/350

**Pull request closed without merging**